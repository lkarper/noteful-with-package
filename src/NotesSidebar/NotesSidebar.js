import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import NotesContext from '../NotesContext/NotesContext';

const NotesSidebar = (props) => {
    return (
        <NotesContext.Consumer>
            {value => {
                const loading = value.notes.length ? false : true;
                const error = value.folderError;
                let folderIdToMatch;
                const noteInContext = value.notes
                    .find(note => note.id === parseInt(props.match.params.noteId));
                if (noteInContext) { 
                    folderIdToMatch = value.notes
                        .find(note => note.id === parseInt(props.match.params.noteId))
                        .folder;
                }

                const folder = folderIdToMatch ? value.folders.find(f => f.id === folderIdToMatch) : null;

                if (folder) {
                    return (
                        <nav className="sidebar">
                            <NavLink
                                className="folder-link"
                                activeClassName="highlighted"
                                to={`/folder/${folder.id}`}
                            >
                                {folder.name}
                            </NavLink>
                            <button 
                                type="buton"
                                onClick={() => props.history.push(`/folder/${folder.id}`)}
                            >
                                Back to folder
                            </button>
                        </nav>
                    );
                }
                
                return (
                    <nav className="sidebar">
                        {loading && !error ? <p>Loading...</p> : 
                            <>
                                <h2>Sorry, could not load your folder.</h2>
                                <p>Check your query for errors, check your network connection and reload the page.</p>
                            </>
                        }
                    </nav>
                );
            }}
        </NotesContext.Consumer>
    );
}

NotesSidebar.defaultProps = {
    match: {
        params: {
            noteId: '',
        },
    },
    history: {
        push: () => {},
    }
}

NotesSidebar.propTypes = {
    match: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
}

export default NotesSidebar;