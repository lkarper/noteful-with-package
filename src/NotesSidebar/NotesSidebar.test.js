import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import NotesSidebar from './NotesSidebar';

const props = {
    match: {
        params: {
            noteId: 'cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1',
        },
    },
}

describe('NotesSideBar Component', () => {
    it('renders the UI without crashing', () => {
        const wrapper = mount(
            <BrowserRouter>
                <NotesSidebar match={props.match} />
            </BrowserRouter>);
        wrapper.unmount();
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(
            <BrowserRouter>
                <NotesSidebar match={props.match} />
            </BrowserRouter>);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });

    it.skip('renders an error when error is passed in context', () => {
        const wrapper = mount(
            <BrowserRouter>
                <NotesSidebar match={props.match} />
            </BrowserRouter>);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    })
});