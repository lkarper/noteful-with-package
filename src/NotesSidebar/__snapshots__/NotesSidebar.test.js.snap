// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesSideBar Component renders an error when error is passed in context 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <NotesSidebar
      match={
        Object {
          "params": Object {
            "noteId": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
          },
        }
      }
    >
      <nav
        className="sidebar"
      >
        <div
          className="folder-load-error"
        >
          <h2>
            Sorry, could not load folders from the server: 
            Could not fetch
            .
          </h2>
          <p>
            Check your network connection and reload the page.
          </p>
        </div>
        <NavLink
          activeClassName="highlighted"
          className="folder-link"
          to="/folder/b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1"
        >
          <Link
            aria-current={null}
            className="folder-link"
            to={
              Object {
                "hash": "",
                "pathname": "/folder/b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
                "search": "",
                "state": null,
              }
            }
          >
            <LinkAnchor
              aria-current={null}
              className="folder-link"
              href="/folder/b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1"
              navigate={[Function]}
            >
              <a
                aria-current={null}
                className="folder-link"
                href="/folder/b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1"
                onClick={[Function]}
              >
                Important
              </a>
            </LinkAnchor>
          </Link>
        </NavLink>
        <button
          onClick={[Function]}
          type="buton"
        >
          Back to folder
        </button>
      </nav>
    </NotesSidebar>
  </Router>
</BrowserRouter>
`;

exports[`NotesSideBar Component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <NotesSidebar
      history={
        Object {
          "push": [Function],
        }
      }
      match={
        Object {
          "params": Object {
            "noteId": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
          },
        }
      }
    >
      <nav
        className="sidebar"
      >
        <h2>
          Sorry, could not load folders.
        </h2>
        <p>
          Check your network connection and reload the page.
        </p>
      </nav>
    </NotesSidebar>
  </Router>
</BrowserRouter>
`;
